### PR TITLE
Stop using scheduled_test.

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

**Comment by nex3** [comment]
@grouma You only need to review the first commit in detail; the second one is mostly just changing boilerplate.

**Comment by nex3** [comment]
This is blocked on #635.

File: test/runner/browser/code_server.dart [hunk 1]
```diff
@@ -4,37 +4,42 @@
...
-/// WebSocket connections.
-///
-/// This uses [ScheduledServer] under the hood, and has similar semantics: its
-/// `handle*` methods all schedule a handler that must be hit before the
-/// schedule will continue.
+/// A class that serves Dart and/or JS code and receives WebSocket connections.
 class CodeServer {
-  /// The underlying server.
-  final ScheduledServer _server;
+  /// The handler for [_server].
```
Comment (grouma) [inline test/runner/browser/code_server.dart]: [just for discussion] Throughout this package you tend to provide doc comments for things that appear to be self evident. Not that I'm strongly opposed but I'm just curious as to why you prefer this style.

File: test/runner/browser/code_server.dart [hunk 2]
```diff
@@ -4,37 +4,42 @@
...
+  final Uri url;
 
-  /// The port of the server, once it's actually instantiated.
-  Future<int> get port => _server.port;
+  static Future<CodeServer> start() async {
+    var server = await HttpMultiServer.loopback(0);
+    var handler = new ShelfTestHandler();
+    shelf_io.serveRequests(server, (request) {
+      if (request.method == "GET" && request.url.path == "favicon.ico") {
+        return new shelf.Response.notFound(null);
```
Comment (grouma) [inline test/runner/browser/code_server.dart]: [optional] Might be worth while adding a simple test favicon to keep the console free of noise.

Comment (nex3) [inline test/runner/browser/code_server.dart]: Can you file an issue for this?

Comment (grouma) [inline test/runner/browser/code_server.dart]: https://github.com/dart-lang/test/issues/637

Comment (nex3) [inline test/runner/browser/code_server.dart]: Thanks!

**Pull request merged**